streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java

```diff
@@ -34,6 +34,60 @@
      */
     Map<MetricName, ? extends Metric> metrics();
 
+    /**
+     * Add a latency, rate and total sensor for a specific operation, which will include the following metrics:
+     * <ol>
+     * <li>average latency</li>
+     * <li>max latency</li>
+     * <li>invocation rate (num.operations / time unit)</li>
+     * <li>total invocation count</li>
+     * </ol>
+     * Whenever a user record this sensor via {@link Sensor#record(double)} etc, it will be counted as one invocation
+     * of the operation, and hence the rate / count metrics will be updated accordingly; and the recorded latency value
+     * will be used to update the average / max latency as well.
+     * The time unit of the latency can be defined by the user.
+     *
+     * Note that you can add more metrics to this sensor after you created it, which can then be updated upon
+     * {@link Sensor#record(double)} calls; but additional user-customized metrics will not be managed by {@link StreamsMetrics}.
```

Comment:
Q: What kind of management does this sentence refer to? When a metric is added to a sensor, the metric is effectively managed by Streams. Am I missing something?

Reply:
Yes I was more concerned about the MBean, but as you remind me, maybe we can also document that `removeSensor()` needs to be called by the user.